src/stateFromElement.js

```diff
@@ -260,6 +276,9 @@ class BlockGenerator {
     if (element.childNodes != null) {
       Array.from(element.childNodes).forEach(this.processNode, this);
     }
+    if (SELF_CLOSING_ELEMENTS.hasOwnProperty(tagName)) {
+      this.processText('~');
```

Comment:
I've gone ahead and merged this since it's not a huge deal. So a followup diff would be appreciated. As for how to do it, maybe something similar to your `SELF_CLOSING_ELEMENTS` except it can be optionally passed in as part of the options which will override the default. The values could be a text char (or char code) instead of `1`. Just a thought.